### PR TITLE
Changed duplication finding algorithm.

### PR DESCRIPTION
**Pull request by reiclone**
Solves #134

### CODE REVIEW

File: imager/daemon.js [hunk 1]
```diff
@@ -548,23 +548,33 @@ function perceptual_hash(src, image, callback) {
...
+	convert(args, src, function(err) {
 		if (err)
 			return callback(Muggle('Hashing error.', err));
-		var bin = path.join(__dirname, 'perceptual');
-		child_process.execFile(bin, [tmp],
-					function (err, stdout, stderr) {
+		var hash='';
+		fs.open(tmp, 'r',function(err,fd) {
+			if (err){
+				fs.unlinkt(tmp);
```
Comment (bakape) [inline imager/daemon.js]: Typo.

File: imager/daemon.js [hunk 2]
```diff
@@ -548,23 +548,33 @@ function perceptual_hash(src, image, callback) {
...
+				return callback(Muggle('Hashing error.', err));
+			}
+			var buffer = new Buffer(1);
+			
+			for(i=0; i<32;i++) {
+				fs.readSync(fd,buffer,0,1,null);
```
Comment (bakape) [inline imager/daemon.js]: In node synchronous I/O should only be used at server boot or shutdown. So you need to use `fs.read()` and put everything after in a callback. Technically, you can stream the file directly from `convert`, but whatever.

File: imager/daemon.js [hunk 3]
```diff
@@ -548,23 +548,33 @@ function perceptual_hash(src, image, callback) {
...
-					function (err, stdout, stderr) {
+		var hash='';
+		fs.open(tmp, 'r',function(err,fd) {
+			if (err){
+				fs.unlinkt(tmp);
+				return callback(Muggle('Hashing error.', err));
+			}
+			var buffer = new Buffer(1);
+			
+			for(i=0; i<32;i++) {
```
Comment (bakape) [inline imager/daemon.js]: Oh, right. `var i=0;` instead of `i=0;`. Javascript potentially pollutes the global scope otherwise.

**Comment by bakape** [comment]
Squashed with slight changes to `fs.readFile()` as 077b148e949300b2f4b3a2c6b61f249b8518e038
Seems fine to you?

**Comment by reiclone** [comment]
Perfect.

**Pull request closed without merging**